prototype2/client/src/engine/mod.rs

```diff
@@ -150,3 +179,124 @@ impl Engine {
     self.tick(dt);
   }
 }
+
+
+// TODO(acmcarther): Move the below (all of it) somewhere else. This is here
+// from the OpenGl service refactoring
+type AspectStorageRead<'a, T> = specs::Storage<T,
+                                               RwLockReadGuard<'a, specs::Allocator>,
+                                               RwLockReadGuard<'a, specs::MaskedStorage<T>>>;
+
+// TODO: Document
+struct SelfRetriever<'a> {
+  own_entity: Option<OwnEntity>,
+  entities: &'a specs::Entities<'a>,
+  synchronized: &'a AspectStorageRead<'a, SynchronizedAspect>,
+  physical: &'a AspectStorageRead<'a, PhysicalAspect>,
+}
+
+impl<'a> SelfRetriever<'a> {
+  pub fn new(own_entity: Option<OwnEntity>,
+             entities: &'a specs::Entities<'a>,
+             synchronized: &'a AspectStorageRead<'a, SynchronizedAspect>,
+             physical: &'a AspectStorageRead<'a, PhysicalAspect>)
+             -> SelfRetriever<'a> {
+
+    SelfRetriever {
+      own_entity: own_entity,
+      entities: entities,
+      synchronized: synchronized,
+      physical: physical,
+    }
+  }
+
+  pub fn find_own_pos(mut self) -> Option<(f32, f32, f32)> {
+    use specs::Join;
+
+    self.own_entity
+      .take()
+      .and_then(|ent| {
+        (self.entities, self.synchronized)
+          .iter()
+          .filter(|&(_, synchro)| {
+            let OwnEntity(ref own_ent) = ent;
+            synchro == own_ent
+          })
+          .next()
+      })
+      .map(|(entity, _)| entity)
+      .and_then(|true_ent| self.physical.get(true_ent))
+      .map(|physical_aspect| physical_aspect.pos.clone())
```

Comment:
I've preferred explicit clones generally just so they're obvious. Regardless, going with convention here and relying on the copy.

In a real perf problem situation I'd do formal profiling anyway and this explicit clone would almost certainly not be helpful.